### ISSUE TITLE
Fit generator slow to converge than model.fit

### ISSUE THREAD

**Issue by hrsmanian** [fchollet/keras#3578]
Hi,
  I am training a 2 hidden layer NN and notice that the convergence is faster when using model.fit than using model.fit_generator. The optimizer is the same in both cases ('sgd') and my batch size is 32  in both cases. val_loss decreases much rapidly when using fit. Is this to be expected? Is val_loss calculated differently for fit_generator?

Thanks in advance

**Comment by bernardohenz** [comment]
I am having the same issue. I've been training exactly the same network, but with model.fit it converges much faster.

**Comment by hrsmanian** [comment]
Hi, what about the validation accuracy? Is it the same for model fit and fit_generator

**Comment by bernardohenz** [comment]
Look at those losses, I am using the same optimizer(Adadelta), batch_size(=32) and same dataset. The 'samples_per_epoch' is the same as the number of images in the dataset (70k). In the first epoch the model.fit has a better loss than fit_generator with 16 epochs. the validation accuracy  follows the loss: 46% in model.fit after 5 epochs, 6% in fit_generator in 16 epochs.


Epochs | Model.fit  |  fit_generator
------------ | ------------- | --------------
1 | 0.01297 | 0.038865
2 | 0.00293 | 0.029753
3 | 0.0019   | 0.028777
4 | 0.00143 | 0.028146
5 | 0.00120 | 0.027444
...|...|...
16 | ?          | 0.026156

**Comment by fchollet** [comment]
The two are functionally identical. Only significant difference is that `fit` defaults to `shuffle=True`; naturally `fit_generator` does not have a `shuffle` option. This is the cause of what you are observing.
